core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitConfig.java

```diff
@@ -166,4 +172,30 @@ public boolean isIncludeMetadataEnabled() {
   public boolean isParallelImplicitPreReadEnabled() {
     return parallelImplicitPreReadEnabled;
   }
+
+  private void validateCrossPartitionScanConfig(DatabaseConfig databaseConfig) {
+    // It might be better to let each storage have metadata (e.g., linearizable cross-partition scan
+    // is supported or not) and check it rather than checking specific storage types. We will
+    // revisit here when supporting metadata management in DistributedStorage.
+    if (databaseConfig.getStorage().equals(MultiStorageConfig.STORAGE_NAME)) {
+      MultiStorageConfig multiStorageConfig = new MultiStorageConfig(databaseConfig);
+      for (Properties props : multiStorageConfig.getDatabasePropertiesMap().values()) {
+        DatabaseConfig c = new DatabaseConfig(props);
+        if (!c.getStorage().equals(JdbcConfig.STORAGE_NAME) && c.isCrossPartitionScanEnabled()) {
+          warnCrossPartitionScan(c.getStorage());
+        }
+      }
+    } else if (!databaseConfig.getStorage().equals(JdbcConfig.STORAGE_NAME)
+        && databaseConfig.isCrossPartitionScanEnabled()) {
+      warnCrossPartitionScan(databaseConfig.getStorage());
+    }
+  }
+
+  private void warnCrossPartitionScan(String storageName) {
+    logger.warn(
+        "Enabling cross-partition scan for {} in production is not recommended "
```

Comment:
[trivial] I imagined what the message would be like (e.g., "Enabling cross-partition scan for cosmos in production ..."), and I think adding quotations might be a bit better.
```suggestion
        "Enabling cross-partition scan for '{}' in production is not recommended "
```

Reply:
Fixed in [7bffcbb](https://github.com/scalar-labs/scalardb/pull/1294/commits/7bffcbb7a942bdcad7ec4c99cdeffa36378226d2). Thank you!